File: nomad-samples/python/basic-live-channel-1.1/live_channel/get_live_channel.py
from constants.project_constants import *
from exceptions.api_exception_handler import *


import requests, json

def get_live_channel(AUTH_TOKEN, CHANNEL_ID):
    # Check for valid parameters
    if (not AUTH_TOKEN or not CHANNEL_ID):
        raise Exception("Get Live Channel: Invalid API call")

    API_URL = f"{ADMIN_URL}/liveChannel/{CHANNEL_ID}"

    # Create header for the request
    HEADERS = {
        'Content-Type': 'application/json',
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    try:
        # Send the request
        RESPONSE = requests.get(API_URL, headers= HEADERS)
        
        
        
        # If not found return None
        if (not RESPONSE.ok):
            raise Exception()
        
        

        return json.loads(RESPONSE.text)


    except:
        api_exception_handler(RESPONSE, f"Get Live Channel with ID {CHANNEL_ID} failed")

File: nomad-samples/admin/js/advanced-live-channel-1.1/live-channel/get-live-channel.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

/**
 * GET Live Channel
 *
 * @param {string} authToken      | Authorization token
 * @param {string} channelId      | The ID of the live channel to get
 *
 * @returns JSON Object containing the requested live channel
 */
export default async function getLiveChannel(authToken, channelId) {
    // Check for valid parameters
    if (!authToken || !channelId) {
        throw new Error("Get Live Channel: Invalid API call");
    }

    // Create header for the request
    const headers = new Headers();
    headers.append("Content-Type", "application/json");
    headers.append("Authorization", `Bearer ${authToken}`);

    // Send the request
    const response = await fetch(`${prjConstants.SERVER_URL}/liveChannel/${channelId}`, {
        method: "GET",
        headers: headers
    });

    // Check for valid response
    if (response) {
        // Check for success
        if (response.ok) {
            // Return the JSON response
            return await response.json();
        }

        // If not found return null
        if (response.status === 404) {
            console.error(`Live Channel with ID ${channelId} was not found`);
            return null;
        }
    }

    await apiExceptionHandler(response, `Get Live Channel with ID ${channelId} failed`);
}
